### PR TITLE
 Day #2: Calculate distance travelled

### PR DESCRIPTION
**Pull request by boarnoah**
Experimented with Rust's support for extending types by implementing traits (so in this case FromStr which is called by string::parse).

Had to do error handling, which turns out to full of boilerplate :(  The current implementation is still wacky, since if I want to print the errors out I need to write even more boilerplate.

Need to do some reading re: modules, ideally the enum + struct I defined would be in a different file in the module (a nested module in rust terms (?))

### CODE REVIEW

File: src/day_two/mod.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+}
+
+impl From<std::io::Error> for CommandParseError {
+    fn from(err: std::io::Error) -> Self {
+        CommandParseError::IoError(err)
+    }
+}
+
+
```
Comment (boarnoah) [inline src/day_two/mod.rs]: Hmm, maybe IDE's formatting isn't as good as I thought lol

File: src/day_two/mod.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    }
+}
+
+struct Command {
+    direction: Direction,
+    distance: u32,
+}
+
+// https://fettblog.eu/rust-enums-wrapping-errors/
+enum CommandParseError {
```
Comment (right-mouse) [inline src/day_two/mod.rs:32]: I'm not too familiar with rust conventions but I personally feel like you could move this and its trait impls to a separate file (errors.rs or something).

**Review by right-mouse — APPROVED** [review]
LGTM

File: src/day_two/mod.rs (hunk 2 above)
Comment (right-mouse) [reply src/day_two/mod.rs:32]: Just read your comment above. We'll worry about this later.

**Pull request merged**